src/iterator.rs

```diff
@@ -593,25 +597,31 @@ mod tokio_support {
         /// use std::io::Error;
         ///
         /// use signal_hook::iterator::Signals;
+        /// use tokio::runtime::current_thread::Runtime;
         /// use tokio::prelude::*;
         ///
         /// fn main() -> Result<(), Error> {
-        ///     let wait_signal = Signals::new(&[signal_hook::SIGUSR1])?
-        ///         .into_async()?
-        ///         .into_future()
-        ///         .map(|sig| assert_eq!(sig.0.unwrap(), signal_hook::SIGUSR1))
-        ///         .map_err(|e| panic!("{}", e.0));
+        ///     let mut runtime = Runtime::new().unwrap();
+        ///
+        ///     let mut signals = Signals::new(&[signal_hook::SIGUSR1])?
+        ///         .into_async()?;
+        ///     let wait_signal = signals
```

Comment:
Did you split the method chain on purpose?

Reply:
I did it because there was the error message about a temporary freed while in use, but I just tried unsplitting the chain and it worked